### PR TITLE
distinguish between randomness from drand vs ticket chain randomness

### PR DESCRIPTION
**Pull request by whyrusleeping**
Ref @nicola's discussion on GetRandomnessFromBeacon and GetRandomnessFromTickets

@anorth edit:
Documenting the behaviour being implemented here before landing it.
- Seal: ticket chain
- PoRep: beacon chain
- Window PoSt: beacon chain

This ties the seal to a particular chain, but not the subsequent PoRep or PoSts.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
FYI @magik6k as this will need to be updated in lotus

File: actors/runtime/runtime.go [hunk 1]
```diff
@@ -57,9 +57,15 @@ type Runtime interface {
 	// Look up the code ID at an actor address.
 	GetActorCodeCID(addr addr.Address) (ret cid.Cid, ok bool)
 
-	// Randomness returns a (pseudo)random byte array drawing from a
+	// GetRandomnessFromBeacon returns a (pseudo)random byte array drawing from a
 	// random beacon at a given epoch and incorporating reequisite entropy
-	GetRandomness(personalization crypto.DomainSeparationTag, randEpoch abi.ChainEpoch, entropy []byte) abi.Randomness
+	GetRandomnessFromBeacon(personalization crypto.DomainSeparationTag, randEpoch abi.ChainEpoch, entropy []byte) abi.Randomness
```
Comment (ZenGround0) [inline actors/runtime/runtime.go:62]: These names are clear 👍

File: actors/builtin/miner/miner_actor.go [hunk 2]
```diff
@@ -1875,8 +1875,8 @@ func getVerifyInfo(rt Runtime, params *SealVerifyStuff) *abi.SealVerifyInfo {
 	err = rt.Message().Receiver().MarshalCBOR(buf)
 	AssertNoError(err)
 
-	svInfoRandomness := rt.GetRandomness(crypto.DomainSeparationTag_SealRandomness, params.SealRandEpoch, buf.Bytes())
-	svInfoInteractiveRandomness := rt.GetRandomness(crypto.DomainSeparationTag_InteractiveSealChallengeSeed, params.InteractiveEpoch, buf.Bytes())
+	svInfoRandomness := rt.GetRandomnessFromTickets(crypto.DomainSeparationTag_SealRandomness, params.SealRandEpoch, buf.Bytes())
+	svInfoInteractiveRandomness := rt.GetRandomnessFromTickets(crypto.DomainSeparationTag_InteractiveSealChallengeSeed, params.InteractiveEpoch, buf.Bytes())
```
Comment (nicola) [inline actors/builtin/miner/miner_actor.go:1879]: This would be `GetRandomnessFromBeacon`

File: actors/runtime/runtime.go (hunk 1 above)
Comment (nicola) [inline actors/runtime/runtime.go:62]: Where do I find the implementation of this in lotus?

Comment (whyrusleeping) [reply actors/runtime/runtime.go:62]: its not there yet, this needs to happen first

File: actors/builtin/miner/miner_actor.go (hunk 2 above)
Comment (whyrusleeping) [reply actors/builtin/miner/miner_actor.go:1879]: I thought we said all the PoRep stuff was using the vrf chain?

Comment (nicola) [reply actors/builtin/miner/miner_actor.go:1879]: No, there is an analysis that shows that we could, but we should rather not do it. It reduces the PoRep security and it changes the way the VRF chain is generated.

Comment (nicola) [reply actors/builtin/miner/miner_actor.go]: I "be the change you want to see"d

File: support/mock/mockrt.go [hunk 3]
```diff
@@ -263,14 +264,37 @@ func (rt *Runtime) GetRandomness(tag crypto.DomainSeparationTag, epoch abi.Chain
...
 		rt.failTest("unexpected get randomness\n"+
 			"         tag: %d, epoch: %d, entropy: %v\n"+
 			"expected tag: %d, epoch: %d, entropy: %v", tag, epoch, entropy, exp.tag, exp.epoch, exp.entropy)
 	}
 	defer func() {
-		rt.expectRandomness = rt.expectRandomness[1:]
+		rt.expectRandomnessBeacon = rt.expectRandomnessBeacon[1:]
+	}()
+	return exp.out
+}
```
Comment (nicola) [inline support/mock/mockrt.go:277]: hey @nikkolasg, can you verify that this is correct according to the spec: https://github.com/filecoin-project/specs/pull/967 ?

File: support/mock/mockrt.go [hunk 4]
```diff
@@ -263,14 +264,37 @@ func (rt *Runtime) GetRandomness(tag crypto.DomainSeparationTag, epoch abi.Chain
...
+	if tag != exp.tag || epoch != exp.epoch || !bytes.Equal(entropy, exp.entropy) {
+		rt.failTest("unexpected get randomness\n"+
+			"         tag: %d, epoch: %d, entropy: %v\n"+
+			"expected tag: %d, epoch: %d, entropy: %v", tag, epoch, entropy, exp.tag, exp.epoch, exp.entropy)
+	}
+	defer func() {
+		rt.expectRandomnessTickets = rt.expectRandomnessTickets[1:]
 	}()
 	return exp.out
 }
```
Comment (nicola) [inline support/mock/mockrt.go:300]: hey @nikkolasg, can you verify that this is correct according to the spec: https://github.com/filecoin-project/specs/pull/967 ?

**Pull request merged**